### PR TITLE
Set default radio button selection, evaluate default component popula…

### LINKED ISSUE #17746: No default selected radio buttons in schema compare dialog
There should be a default selected source and target type. Probably caused by #17679

actual:
image

### PR DESCRIPTION
**Pull request by Benjin**
Fixed #17746 

Two bugs fixed:
* No default selection was being set if the source or target weren't provided (both valid scenarios)
* The displayed components were being populated off of the input source and target, rather than the current selection (which would reflect the input, except when a default selection was made

### CODE REVIEW

File: extensions/schema-compare/src/dialogs/schemaCompareDialog.ts [hunk 1]
```diff
@@ -610,16 +602,21 @@ export class SchemaCompareDialog {
 			this.dialog.okButton.enabled = await this.shouldEnableOkayButton();
 		});
 
-		// if target is currently a db, show it in the server and db dropdowns
-		if (this.schemaCompareMainWindow.targetEndpointInfo && this.schemaCompareMainWindow.targetEndpointInfo.endpointType === mssql.SchemaCompareEndpointType.Database) {
+		if (this.schemaCompareMainWindow.targetEndpointInfo) {
+			if (this.schemaCompareMainWindow.targetEndpointInfo.endpointType === mssql.SchemaCompareEndpointType.Database) {
+				databaseRadioButton.checked = true;
```
Comment (kisantia) [inline extensions/schema-compare/src/dialogs/schemaCompareDialog.ts:607]: can you also update these radio button variable names to start with target so it's more clear while you're here?

**Review by kisantia — APPROVED** [review]

File: extensions/schema-compare/src/dialogs/schemaCompareDialog.ts [hunk 2]
```diff
@@ -525,16 +512,21 @@ export class SchemaCompareDialog {
 			this.dialog.okButton.enabled = await this.shouldEnableOkayButton();
 		});
 
-		// if source is currently a db, show it in the server and db dropdowns
-		if (this.schemaCompareMainWindow.sourceEndpointInfo && this.schemaCompareMainWindow.sourceEndpointInfo.endpointType === mssql.SchemaCompareEndpointType.Database) {
+		if (this.schemaCompareMainWindow.sourceEndpointInfo) {
```
Comment (chlafreniere) [inline extensions/schema-compare/src/dialogs/schemaCompareDialog.ts:515]: any reason why you aren't using a switch statement consistently in this PR?

Comment (Benjin) [reply extensions/schema-compare/src/dialogs/schemaCompareDialog.ts:515]: Hmm... not particularly.  Lemme massage it a bit.

File: extensions/schema-compare/src/dialogs/schemaCompareDialog.ts (hunk 1 above)
Comment (Benjin) [reply extensions/schema-compare/src/dialogs/schemaCompareDialog.ts:607]: Done.  It's also odd that the source buttons are member-scoped, but the target buttons are local-scoped.  Do you know if there's a room for that?  I'm not spotting one in the code (aside from some changeable order-of-operations)

**Pull request merged**